intelmq/bots/parsers/microsoft/parser_ctip.py

```diff
@@ -291,6 +291,9 @@ def parse_azure(self, line, report):
                 if payload_protocol:
                     # needs to overwrite a field previously parsed and written
                     event.add('protocol.application', payload_protocol, overwrite=True)  # "HTTP/1.1", save additionally
+            elif key == 'Payload.domain':
+                if not FQDN.is_valid(value):
+                    continue
```

Comment:
Can we be more strict here to prevent accidental loss of data? Just in case that the data some day does not only put the IP address there, but actually invalid domains or so. Or different IP addresses, which you might not want to miss.
```suggestion
            elif key == 'Payload.domain':
                # Sometimes the destination address is also given as domain, ignore it here as we already save it as destination.ip (see https://github.com/certtools/intelmq/pull/2144)
                if not FQDN.is_valid(value) and value == line.get('Payload.serverIp'):
                    continue
```